backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+	}
+
+	_, err := d.clientClient.CreateOrchestration(ctx, &dtmbprotos.CreateOrchestrationRequest{
+		OrchestrationId: executionStartedEvent.GetOrchestrationInstance().GetInstanceId(),
+		Name:            executionStartedEvent.GetName(),
+		Version:         executionStartedEvent.GetVersion().GetValue(),
+		Input:           []byte(executionStartedEvent.GetName()),
+		StartAt: &dtmbprotos.Delay{
+			Delayed: &dtmbprotos.Delay_Time{
+				Time: executionStartedEvent.GetScheduledStartTimestamp(),
+			},
+		},
+		IdReusePolicy: &dtmbprotos.OrchestrationIDReusePolicy{ // is this correct?
+			RuntimeStatus: nil, // []dtmbprotos.OrchestrationStatus{}, // what do I put here?
+			// Action:        dtmbprotos.OrchestrationIDReusePolicy_ERROR, // this is the default value
+		},
+	})
+
+	return err
+}
+
+// AddNewEvent adds a new orchestration event to the specified orchestration instance.
+func (d *dtmb) AddNewOrchestrationEvent(ctx context.Context, id api.InstanceID, event *backend.HistoryEvent) error {
+	var err error
+	switch typedEvent := event.GetEventType().(type) {
+	case *protos.HistoryEvent_EventRaised:
+
+		req := dtmbprotos.RaiseEventRequest{
+			OrchestrationId: string(id),
+			Name:            typedEvent.EventRaised.GetName(),
+			Input:           []byte(typedEvent.EventRaised.GetInput().GetValue()), // or should this be backend.MarshalHistoryEvent(event)?
+		}
+
+		_, err = d.clientClient.RaiseEvent(context.Background(), &req)
+
+	case *protos.HistoryEvent_ExecutionTerminated:
+		// this is a terminal event, so we can evict the cache
+		d.orchestrationHistoryCache.EvictCacheForOrchestrationID(string(id))
+		err = fmt.Errorf("not implemented in protos")
+	case *protos.HistoryEvent_ExecutionResumed:
+		err = fmt.Errorf("not implemented in protos")
+	case *protos.HistoryEvent_ExecutionSuspended:
+		err = fmt.Errorf("not implemented in protos")
+	default:
+		err = fmt.Errorf("unsupported event type in AddNewOrchestrationEvent: %v", event.GetEventType())
+	}
+
+	return err
+}
+
+func (d *dtmb) getOrchestrationHistory(ctx context.Context, orchestrationID string) ([]*dtmbprotos.Event, error) {
+	// look up cached history events and request the rest from the server
+	cachedEvents := d.orchestrationHistoryCache.GetCachedHistoryEventsForOrchestrationID(orchestrationID)
+
+	var historyRequest dtmbprotos.GetOrchestrationHistoryRequest
+	if cachedEvents == nil {
+		historyRequest = dtmbprotos.GetOrchestrationHistoryRequest{
+			OrchestrationId: orchestrationID,
+		}
+	} else {
+		historyRequest = dtmbprotos.GetOrchestrationHistoryRequest{
+			OrchestrationId:        orchestrationID,
+			LastItemSequenceNumber: cachedEvents[len(cachedEvents)-1].GetSequenceNumber(),
+			LastItemEventHash:      cachedEvents[len(cachedEvents)-1].GetEventHash(),
+		}
+	}
+
+	res, err := d.workerClient.GetOrchestrationHistory(
+		ctx,
+		&historyRequest,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	events := res.GetEvent()
```

Comment:
That would require a change in the proto cc @ItalyPaleAle

Reply:
I will change that in the next iterations.

In the meanwhile, feel free to change the `proto` in this PR. Even if the name is different, they are compatible on the wire, so it will still work.

Reply:
Ok. I've made the change in the current open PR.